connector/src/it/scala/com/datastax/spark/connector/datasource/CassandraCatalogNamespaceSpec.scala

```diff
@@ -0,0 +1,104 @@
+package com.datastax.spark.connector.datasource
+
+import org.scalatest.concurrent.Eventually._
+import org.apache.spark.sql.catalyst.analysis.{NamespaceAlreadyExistsException, NoSuchNamespaceException}
+import scala.collection.JavaConverters._
+
+class CassandraCatalogNamespaceSpec extends CassandraCatalogSpecBase {
+
+  "A Cassandra Catalog Namespace Support" should "initialize successfully" in {
+    spark.sessionState.catalogManager.currentCatalog.name() should include("Catalog cassandra")
+  }
+
+  it should "list available keyspaces" in {
+    val currentKeyspaces = spark.sql("SHOW DATABASES").collect().map(_.getString(0))
+    val existingKeyspaces = getMetadata().getKeyspaces().keySet().asScala.map(_.asInternal())
+    currentKeyspaces should contain allElementsOf(existingKeyspaces)
+
+  }
+
+  it should "list a keyspaces in a keyspace"  in {
+    spark.sql(s"CREATE DATABASE IF NOT EXISTS $defaultKs WITH DBPROPERTIES (class='SimpleStrategy',replication_factor='5')")
+    waitForKeyspaceToExist(defaultKs, true)
+    //Using Catalog and DefaultKS triggers listNamespaces(namespace) pathway
+    val result = spark.sql(s"SHOW NAMESPACES FROM $catalogName.$defaultKs").collect
+    result shouldBe empty
+  }
+
+  it should "give no keyspace found when looking up a keyspace which doesn't exist" in {
+    intercept[NoSuchNamespaceException]{
+      spark.sql(s"SHOW NAMESPACES FROM $catalogName.fakenotreal").collect
+    }
+  }
+
+  it should "be able to create a new keyspace" in {
+    dropKeyspace(defaultKs)
+    waitForKeyspaceToExist(defaultKs, false)
+    spark.sql(s"CREATE DATABASE $defaultKs WITH DBPROPERTIES (class='SimpleStrategy',replication_factor='5')")
```

Comment:
Nit: worth validating that defaultKs no longer exists here?  I'm not sure it buys you much but _might_ be useful.

Reply:
That's what the line 36 does, and if it DID exist this would actually fail with NamespaceExistsException :)